packages/gateway-client/src/worker/p2p-client/stream-factory.ts

```diff
@@ -139,4 +222,27 @@ export class StreamFactory {
     public async getStream(protocol = '/samizdapp-proxy') {
         return this.makeStream(protocol);
     }
+
+    public async getProxyStream() {
```

Comment:
I'm not sure I like coupling a specific protocol with different stream logic this tightly. I'd like to see a separation between the `samizdapp-proxy` protocol, the concept of a stream pool, and the concept of a request stream?

Why not introduce the pooling concept to `getStream()`? You could add a second optional options parameter to it with a `pooled = false` property.

The above would result in:
 - `getStream()` with `protocol` and `options` (pooled = false) params
 - `getProxyStream()` is renamed to `getRequestStream()` that takes same two params
   - Both of these params are passed to `new RequestStream()` (which, in line of the above comment, in turn calls `getStream()`)

If you want to introduce *some* coupling between the proxy protocol and the request stream logic, there are some additional patterns you could implement.